### PR TITLE
extend search table with trusted and owners info

### PR DESCRIPTION
**Pull request by YuliiaKovalova**
## Problem

The current search output doesn't provide much information about a package.
These changes add details about package owners and highlight if the package is verified https://learn.microsoft.com/en-us/nuget/nuget-org/id-prefix-reservation.

image

related to 
https://github.com/dotnet/templating/issues/5860
https://github.com/dotnet/templating/issues/3456

### CODE REVIEW

**Comment by YuliiaKovalova** [comment]
@baronfel , could you advise regarding column naming here?

File: src/Cli/Microsoft.TemplateEngine.Cli/LocalizableStrings.resx [hunk 1]
```diff
@@ -393,6 +393,9 @@ Run 'dotnet {1} --show-aliases' with no args to show all aliases.</value>
   <data name="ColumnNameAuthor" xml:space="preserve">
     <value>Author</value>
   </data>
+  <data name="ColumnNamePackageNameAndOwners" xml:space="preserve">
+    <value>Package Name / Owners</value>
```
Comment (vlada-shubina) [inline src/Cli/Microsoft.TemplateEngine.Cli/LocalizableStrings.resx:397]: please consider adding comments for translators

File: src/Cli/Microsoft.TemplateEngine.Cli/TabularOutput/TemplateGroupTableRow.cs [hunk 2]
```diff
@@ -11,6 +11,8 @@ internal struct TemplateGroupTableRow
     {
         internal string Author { get; set; }
 
+        internal string PackageAuthors { get; set; }
```
[the remark this answers is not in the document]
Comment (YuliiaKovalova) [reply src/Cli/Microsoft.TemplateEngine.Cli/TabularOutput/TemplateGroupTableRow.cs]: done

File: src/Cli/Microsoft.TemplateEngine.Cli/TabularOutput/TabularOutput.cs [hunk 3]
```diff
@@ -310,7 +317,7 @@ internal ColumnDefinition(TabularOutputSettings settings, string? header, Func<T
 
             internal bool ShrinkIfNeeded { get; }
 
-            internal bool RightAlign { get; }
+            internal bool CenterAlign { get; }
```
[the remark this answers is not in the document]
Comment (YuliiaKovalova) [reply src/Cli/Microsoft.TemplateEngine.Cli/TabularOutput/TabularOutput.cs]: done

File: src/Cli/Microsoft.TemplateEngine.Cli/LocalizableStrings.resx (hunk 1 above)
Comment (YuliiaKovalova) [reply src/Cli/Microsoft.TemplateEngine.Cli/LocalizableStrings.resx:397]: done

File: src/Cli/Microsoft.TemplateEngine.Cli/LocalizableStrings.resx [hunk 4]
```diff
@@ -393,6 +393,10 @@ Run 'dotnet {1} --show-aliases' with no args to show all aliases.</value>
...
+    <comment>information about package: it's name and owner</comment>
```
Comment (vlada-shubina) [inline src/Cli/Microsoft.TemplateEngine.Cli/LocalizableStrings.resx:398]:
```suggestion
    <comment>information about NuGet package: it's name and owner</comment>
```

File: src/Cli/Microsoft.TemplateEngine.Cli/LocalizableStrings.resx [hunk 5]
```diff
@@ -814,4 +818,8 @@ The header is followed by the list of parameters and their errors (might be seve
   <data name="UnexpectedResult" xml:space="preserve">
     <value>Template generation ended with unexpected result: '{0}'. Details: {1}</value>
   </data>
+  <data name="ColumnNameTrusted" xml:space="preserve">
+    <value>Trusted</value>
+    <comment>information about package origin; if a package has PrefixReserved indicator </comment>
```
Comment (vlada-shubina) [inline src/Cli/Microsoft.TemplateEngine.Cli/LocalizableStrings.resx:823]:
```suggestion
    <comment>information about NuGet package origin; if a package has PrefixReserved indicator </comment>
```

File: src/Cli/Microsoft.TemplateEngine.Cli/TemplateInvoker.cs [hunk 6]
```diff
@@ -47,6 +47,12 @@ internal async Task<NewCommandStatus> InvokeTemplateAsync(TemplateCommandArgs te
             string? framework = isMicrosoftAuthored ? TelemetryHelper.PrepareHashedChoiceValue(templateToRun, templateParameters, "Framework") : null;
             string? auth = isMicrosoftAuthored ? TelemetryHelper.PrepareHashedChoiceValue(templateToRun, templateParameters, "auth") : null;
             string? templateName = Sha256Hasher.HashWithNormalizedCasing(templateToRun.Identity);
+            string? templateShortNames = templateToRun.ShortNameList.Any() ? Sha256Hasher.HashWithNormalizedCasing(string.Join(',', templateToRun.ShortNameList)) : null;
+
+            using TemplatePackageManager templatePackageManager = new(_environmentSettings);
+            var templatePackage = await templateArgs.Template.GetManagedTemplatePackageAsync(templatePackageManager, cancellationToken).ConfigureAwait(false);
+            string? packageName = string.IsNullOrEmpty(templatePackage?.Identifier) ? null : Sha256Hasher.HashWithNormalizedCasing(templatePackage.Identifier);
+            string? packageVersion = string.IsNullOrEmpty(templatePackage?.Version) ? null : Sha256Hasher.HashWithNormalizedCasing(templatePackage.Version);
```
Comment (vlada-shubina) [inline src/Cli/Microsoft.TemplateEngine.Cli/TemplateInvoker.cs:55]: it's better to do those changes in separate PR next time

**Review by vlada-shubina — APPROVED** [review]
